mono/mini/interp/transform.c

```diff
@@ -4450,8 +4452,13 @@ mono_interp_transform_method (InterpMethod *imethod, ThreadContext *context, Int
 					return mono_error_convert_to_exception (&error);
 				}
 				mono_class_init (m->klass);
-				if (!mono_class_is_interface (m->klass))
-					mono_class_vtable (domain, m->klass);
+				if (!mono_class_is_interface (m->klass)) {
+					mono_class_vtable_full (domain, m->klass, &error);
+					if (!is_ok (&error)) {
+						g_free (is_bb_start);
+						return mono_error_convert_to_exception (&error);
```

Comment:
not exactly obvious, but it should also call `mono_metadata_free_mh (header);`.  On the other hand, how often are we going to hit this case?